apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php

```diff
@@ -77,17 +77,42 @@ private function register_patterns() {
 
 		foreach ( (array) $block_patterns as $pattern ) {
 			foreach ( (array) $pattern['categories'] as $slug => $category ) {
-				// Temporarily skip the 'featured' category so that we can expose it at another time.
-				if ( 'featured' !== $slug ) {
-					$pattern_categories[ $slug ] = $category['title'];
-				}
+				$pattern_categories[ $slug ] = array( 'label' => $category['title'] );
 			}
 		}
 
-		// Order categories alphabetically and register them.
-		asort( $pattern_categories );
-		foreach ( (array) $pattern_categories as $slug => $label ) {
-			register_block_pattern_category( $slug, array( 'label' => $label ) );
+		// Unregister existing categories so that we can insert them in the desired order (alphabetically).
+		$existing_categories = array();
+		foreach ( \WP_Block_Pattern_Categories_Registry::get_instance()->get_all_registered() as $existing_category ) {
+			$existing_categories[ $existing_category['name'] ] = $existing_category;
+			unregister_block_pattern_category( $existing_category['name'] );
+		}
+
+		$pattern_categories = array_merge( $pattern_categories, $existing_categories );
+
+		// Order categories alphabetically by their label.
+		uasort(
+			$pattern_categories,
+			function ( $a, $b ) {
+				return strnatcasecmp( $a['label'], $b['label'] );
+			}
+		);
+
+		// Move the Featured category to be the first category.
+		if ( isset( $pattern_categories['featured'] ) ) {
+			$featured_category = $pattern_categories['featured'];
```

Comment:
just a thought, not critical, i think the array union operator might be able to simplify this, eg.

```php
$pattern_categories = array( 'featured' => $featured_category), + $pattern_categories;
```
so no need to find or unset the featured one first - haven't actually testing it though to see if it does work

Reply:
Oh nice, thanks for the suggestion @glendaviesnz, I didn't realise the union operator would preserve the order. I'll give it a shot!

Reply:
It works nicely! I've kept the `isset` check in there so that we're not adding in a `featured` category if it ever drops off the source site (and so that nothing breaks when we're testing with separate arbitrary block patterns source sites).